frame/support/src/inherent.rs

```diff
@@ -64,21 +69,29 @@ pub trait ProvideInherent {
 	/// included in the block by its author. Whereas the second parameter represents the inherent
 	/// data that the verifying node calculates.
 	///
-	/// NOTE: A block can contains multiple inherent.
+	/// This is intended to allow for checks that cannot be done within the runtime such as, e.g.,
+	/// the timestamp.
+	///
+	/// NOTE: A block can contain multiple inherents.
+	///
+	/// # Warning
+	///
+	/// This check is not guaranteed to be run as part of consensus and cannot be relied upon for
+	/// security. Run security relevant checks in [`Self::create_inherent`].
```

Comment:
security checks cannot be put in create_inherent AFAIK.
I don't really know the nature of the check that `check_inherent` provide, but it is important for security reason otherwise there is no point to it, no ?
Some check for the validity timestamp is done in `check_inherent`.

As far as I understand this: It is what other validator will use the check the validity of the inherents provided in comparison to some inherent data that the validator will provide.
For instance, a validator will check that the timestamp is not too much in the future compared to what he considers to be the present. (I.e. compared to the time he provided in his inherent data)